### PR TITLE
Log pipeline test failures

### PR DESCRIPTION
**Pull request by zhenmichael**
PR to replace https://github.com/microsoft/FluidFramework/pull/14029 due to build failures.
updated test:mocha:report script in root package.json to use a multi reporter which generates both json and junit reports. Results from json reports are printed in the console of a pipeline run.

image

### CODE REVIEW

File: scripts/report-parser.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Parsing script which finds all test reports ending with "junit-report.json" 
+ * in a given directory and prints failed tests into the console
+ */
+
+var fs = require('fs');
+var util = require('util');
+var path = require('path');
+
+var directory = process.argv.slice(2)[0];
```
Comment (scottn12) [inline scripts/report-parser.js:15]:
```suggestion
const directory = process.argv.slice(2)[0];
```

File: scripts/report-parser.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        getJunitTestReports(absolute);
+      } else if (/junit-report.json$/.test(absolute)) {
+          files.push(absolute);
+      }
+    }
+  }
+
+  getJunitTestReports(directory);
+
+let output = [];
```
Comment (scottn12) [inline scripts/report-parser.js:33]: nit: const
```suggestion
const output = [];
```

File: scripts/report-parser.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+for(const filename of files) {
+    const jsonData = fs.readFileSync(filename,  'utf8');
+
+    const failedTests = JSON.parse(jsonData).failures;
+
+    if(failedTests.length > 0) output.push(failedTests)
```
Comment (scottn12) [inline scripts/report-parser.js:39]: nit
```suggestion
    if(failedTests.length > 0) output.push(failedTests);
```

**Review by scottn12 — APPROVED** [review]
Left a few nits on the JS files, but overall looks good!

File: package.json [hunk 4]
```diff
@@ -78,7 +78,7 @@
 		"test:memory-profiling:report": "lerna run test:memory-profiling:report --stream --no-bail --no-sort -- -- --reporterOptions reportDir=./memoryProfilingOutput",
 		"test:mocha": "lerna run test:mocha --stream --no-bail --no-sort -- --color",
 		"test:mocha:bail": "lerna run test:mocha --stream",
-		"test:mocha:report": "lerna run test:mocha --stream --no-bail --no-sort -- --timeout 4s --reporter xunit --reporter-option output=nyc/mocha-junit-report.xml",
+		"test:mocha:report": "lerna run test:mocha:multireport --stream --no-bail --no-sort -- -- --timeout 4s",
```
Comment (anthony-murphy) [inline package.json:81]: this should be reverted

**Pull request merged**